### ISSUE TITLE
owl-item width:0px

### ISSUE THREAD

**Issue by giolongo** [vitalii-andriiovskyi/ngx-owl-carousel-o#105]
Item's with tag p has width of 0px
![image](https://user-images.githubusercontent.com/32745126/87442556-b9e97b80-c5f4-11ea-9659-da2a7ef090a4.png)

**Comment by Superlukec** [comment]
I can confirm this behaviour. 

To fix this, I need to trigger 'resize' event. After that it works normally.

**Comment by dinopascale** [comment]
> I can confirm this behaviour.
> 
> To fix this, I need to trigger 'resize' event after first initialization. After that it works normally.

Hi, maybe a noob question, but how do you retrigger 'resize' event after first initialization? I'm trying to resolve this issue for a while, but I have not found a right approach yet. Thank you

**Comment by giolongo** [comment]
> I can confirm this behaviour.
> To fix this, I need to trigger 'resize' event after first initialization. After that it works normally.

I "solved" this with a littel workaround.
On carouselSlide element I setted attribute [width]="number" and on customOptions I setted autoWidth attribute at true.

I don't kwnow if this is solution is correct, but in my personal case, I solved thanks of it.

**Comment by Superlukec** [comment]
You can trigger the 'resize' event like that:

`window.dispatchEvent(new Event('resize'));`

In my case I do it like:

`setTimeout(() => {            
   window.dispatchEvent(new Event('resize'));          
}, 500)`

**Comment by vitalii-andriiovskyi** [comment]
@giolongo 

The carousel always inherits the width of its parent DOM-element. If it has the width of 0 at the moment the carousel starts rendering, `.owl-stage` and `.owl-item` get the width of `0`, Check the width of parent DOM-element at the moment the carousel starts rendering itself. Also this issue could be caused by your CSS

**Comment by harikrishnanng** [comment]
Hi @vitalii-andriiovskyi,
I'm also facing this same issue. I have 2 carousels in separate components on the same page. One of them is a product slider and the other is a banner slider. The product slider works fine but the banner slider only shows when the 'resize' event occurs. For now, all the data is static. Even then this issue occurs.

> "ngx-owl-carousel-o": "^4.0.0",
> "angular":"10.0.9"

**Comment by vitalii-andriiovskyi** [comment]
@harikrishnanng 

find out the width of the DOM-element you put the banner slide in at the moment the carousel gets rendered. 
Check CSS of that element. `display: flex` could lead to this issue.

**Comment by harikrishnanng** [comment]
@vitalii-andriiovskyi Thanks mate. `display: flex` was the issue. In my case, the parent of the div containing the `<owl-carousel-o>` had that prop.

**Comment by Vivekdubey21** [comment]
> @harikrishnanng
> 
> find out the width of the DOM-element you put the banner slide in at the moment the carousel gets rendered.
> Check CSS of that element. `display: flex` could lead to this issue.

This works for me and i think this was the reason. Thanks for the solution.

**Comment by vivekdubey19** [comment]
> @harikrishnanng
> 
> find out the width of the DOM-element you put the banner slide in at the moment the carousel gets rendered.
> Check CSS of that element. `display: flex` could lead to this issue.

This works for me and i think this was the reason. Thanks for the solution.

**Comment by SandeepBanerjee** [comment]
> > @harikrishnanng
> > find out the width of the DOM-element you put the banner slide in at the moment the carousel gets rendered.
> > Check CSS of that element. `display: flex` could lead to this issue.
> 
> This works for me and i think this was the reason. Thanks for the solution.

yes its workign for me as I have added that owl-corousal into a row class,
